std/algorithm/searching.d

```diff
@@ -3244,6 +3244,162 @@ auto maxElement(alias map = "a", Range, RangeElementType = ElementType!Range)
     assert(arr2d.maxElement!"a[1]" == arr2d[1]);
 }
 
+/**
+Iterates the passed range and selects the min and max element with `less`.
+If the extreme element occurs multiple time, the first occurrence will be
+returned.
+This function is more efficient than calling both $(LREF minElement) and
+$(LREF maxElement).
```

Comment:
Could go into a little more detail here. I think you should add something like 

> This function is more efficient than calling both $(LREF minElement) and $(LREF maxElement) for one range because this function only requires one scan of the range, whereas the former takes two. Also, calling both $(LREF minElement) and $(LREF maxElement) on the same range would require it to be a forward range.